src/Symfony/Bridge/Doctrine/CacheWarmer/ProxyCacheWarmer.php

```diff
@@ -44,7 +44,7 @@ public function isOptional()
     /**
      * {@inheritdoc}
      */
-    public function warmUp($cacheDir)
+    public function warmUp($cacheDir): void
```

Comment:
I personally don't like "void": very low value (we don't add `@return void` for the same reason IMHO)
BC breaking change of course
and only my POV, so up for discussion of course :)

Reply:
in general :+1: for void; we imply mixed otherwise. So in terms of closing the gates, and to follow @stof 's [comment](https://github.com/symfony/symfony/issues/24882#issuecomment-343153260), i agree with;

> we should indeed add types whenever we can

so we can safely assume no type means mixed. But im not sure https://wiki.php.net/rfc/mixed-typehint changes our game.

Reply:
The "void" return type will help PHP to execute code faster. PHP 7.2 will have better code optimizer. Any type hint will reduce the amount of time to execute code since no need to detect type.

Reply:
Source? As far as I know, type hinting in PHP has negative performance impact. The aspect I personally don't like about void return type hint is that it will prevent subclasses from making fluent interfaces. Other than that, sure, it makes code more clear.

Reply:
@ostrolucky http://nikic.github.io/pdf/cc17_static_optimization.pdf